daemon/logger/syslog/syslog.go

```diff
@@ -0,0 +1,61 @@
+package syslog
+
+import (
+	"log/syslog"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/docker/docker/daemon/logger"
+)
+
+type Syslog struct {
+	writer *syslog.Writer
+	tag    string
+	mu     sync.Mutex
+}
+
+func New(tag string) (logger.Logger, error) {
+	return &Syslog{
+		writer: nil,
+		tag:    tag,
+	}, nil
+}
+
+func (s *Syslog) Log(msg *logger.Message) error {
+	s.mu.Lock()
```

Comment:
You still will need lock on whole `Log` if you want to replace `s.writer` in it :( I think probably failing on `Dial` is good idea, maybe with some retry/timeout logic. Because there are plenty of cases when failure better then loss of logs.